src/main/scala/org/camunda/feel/impl/parser/FeelParser.scala

```diff
@@ -118,7 +113,7 @@ object FeelParser extends JavaTokenParsers {
   private lazy val expression8 = functionInvocation | builtinFunctionInvocation | filteredExpression9
   // 2 i)
   private lazy val expression9 =
-    not(dateTimeWords) ~> name ^^ (n => Ref(List(n))) |
+    not(dateTimeLiteral) ~> name ^^ (n => Ref(List(n))) |
```

Comment:
`dateTimeLiteral` seems to be defined for the date-time literals OR it is a failure. 

```scala
// Line 211 of this file
private lazy val dateTimeLiteral: Parser[Exp] =
    "date" ~ "(" ~> stringLiteralWithQuotes <~ ")" ^^ parseDate |
      "time" ~ "(" ~> stringLiteralWithQuotes <~ ")" ^^ parseTime |
      "date and time" ~ "(" ~> stringLiteralWithQuotes <~ ")" ^^ parseDateTime |
      "duration" ~ "(" ~> stringLiteralWithQuotes <~ ")" ^^ parseDuration |
      failure("expected date time literal")
```

Wouldn't that cause a problem here?

Reply:
Nope. All tests pass :wink: 

The failure is defined in the grammar to produce a nicer failure message. It would be returned if the parser can not backtrack from the `dateTimeLiteral` parser (i.e. if there are no other parsers that match the token).

Does this make sense for you?

Reply:
Ah okay. But why then does this work fine? Is it only because of backtracking? expression9 is used somewhere and the parser can still backtrack to a next part of the grammar definition; or is this `not()` method able to deal with a failure as a `not(false)`? Also does the failure method return `Parser[Exp]` type?

Reply:
Yes. Yes. And no :sweat_smile: 

Let's have a quick chat :wink:

Reply:
Awesome! Then LGTM. Have a 🌮